### PR TITLE
transactions - tx history for contract account

### PR DESCRIPTION
**Pull request by frankiebee**

### CODE REVIEW

File: app/scripts/controllers/transactions/index.js [hunk 1]
```diff
@@ -632,13 +632,24 @@ class TransactionController extends EventEmitter {
...
+
+    if (isContractAccount) {
+      selectedAddressTxList.concat(this.txStateManager.getFilteredTxList({
+        from: this.accountsController.currentContractInstance.controllingAccount,
+        to: this.getSelectedAddress(),
+        metamaskNetworkId: this.getNetwork(),
+      }))
+    }
+
+    this.memStore.updateState({ unapprovedTxs, selectedAddressTxList: {} })
```
Comment (frankiebee) [inline app/scripts/controllers/transactions/index.js]: wtf is this frankie

File: app/scripts/controllers/transactions/index.js [hunk 2]
```diff
@@ -632,13 +632,24 @@ class TransactionController extends EventEmitter {
...
     const selectedAddressTxList = this.txStateManager.getFilteredTxList({
       from: this.getSelectedAddress(),
       metamaskNetworkId: this.getNetwork(),
     })
-    this.memStore.updateState({ unapprovedTxs, selectedAddressTxList })
+
+    if (isContractAccount) {
+      selectedAddressTxList.concat(this.txStateManager.getFilteredTxList({
+        from: this.accountsController.currentContractInstance.controllingAccount,
+        to: this.getSelectedAddress(),
```
Comment (frankiebee) [inline app/scripts/controllers/transactions/index.js]: needs to be the contract account

File: app/scripts/controllers/transactions/index.js [hunk 3]
```diff
@@ -632,13 +632,24 @@ class TransactionController extends EventEmitter {
...
     const unapprovedTxs = this.txStateManager.getUnapprovedTxList()
     const selectedAddressTxList = this.txStateManager.getFilteredTxList({
       from: this.getSelectedAddress(),
       metamaskNetworkId: this.getNetwork(),
     })
-    this.memStore.updateState({ unapprovedTxs, selectedAddressTxList })
+
+    if (isContractAccount) {
+      selectedAddressTxList.concat(this.txStateManager.getFilteredTxList({
+        from: this.accountsController.currentContractInstance.controllingAccount,
```
Comment (frankiebee) [inline app/scripts/controllers/transactions/index.js]: needs to be the controlling account

**Comment by jennypollack** [comment]
thanks for this! adding it slightly differently with the tx type in another PR

**Pull request closed without merging**